components/email/Composer.tsx

```diff
@@ -93,31 +105,37 @@ export const EmailComposer = ({ did }: { did: string }) => {
         .processSync(content)
         .toString()
 
-      await toast.promise(
-        labelerAgent.api.tools.ozone.moderation.emitEvent({
-          event: {
-            $type: MOD_EVENTS.EMAIL,
-            comment,
-            subjectLine: subject,
-            content: htmlContent,
-          },
-          subject: { $type: 'com.atproto.admin.defs#repoRef', did },
-          createdBy: labelerAgent.assertDid,
-        }),
-        {
-          pending: 'Sending email...',
-          success: {
-            render() {
-              return 'Email sent to user'
+      const event = {
+        $type: MOD_EVENTS.EMAIL,
+        comment,
+        subjectLine: subject,
+        content: htmlContent,
+      }
+      if (handleSubmit) {
+        await handleSubmit(event)
+      } else {
+        await toast.promise(
+          labelerAgent.tools.ozone.moderation.emitEvent({
+            event,
+            createdBy: labelerAgent.assertDid,
+            subject: { $type: 'com.atproto.admin.defs#repoRef', did },
+          }),
+          {
+            pending: 'Sending email...',
+            success: {
+              render() {
+                return 'Email sent to user'
+              },
             },
-          },
-          error: {
-            render() {
-              return 'Error sending email'
+            error: {
+              render() {
+                return 'Error sending email'
+              },
             },
           },
-        },
-      )
+        )
```

Comment:
Could we make `handleSubmit` non-optional and use this implementation as `handleSubmit` wherever this is used ? Alternatively, a simpler re-factor would be to create a new component that provides this `handleSubmit` implementation:

```tsx
export function EmailComposer ({ did }: { did: string }) {
  return <EmailComposerBase
    did={did}
    handleSubmit={
      // tools.ozone.moderation.emitEvent + toast here
    }
    />
})

export function EmailComposerBase ({
  did,
  replacePlaceholders = true,
  handleSubmit,
}: {
  did?: string
  replacePlaceholders?: boolean
  handleSubmit?: (emailData: EmailComposerData) => Promise<void>
}) {

}
```